src/ngx_http_lua_ssl_certby.c

```diff
@@ -854,6 +854,43 @@ ngx_http_lua_ffi_ssl_server_name(ngx_http_request_t *r, char **name,
 }
 
 
+int
+ngx_http_lua_ffi_ssl_server_port(ngx_http_request_t *r, unsigned int *port,
+    char **err)
+{
+    ngx_ssl_conn_t          *ssl_conn;
+    ngx_connection_t        *c;
+    ngx_uint_t               p;
+
+    if (r->connection == NULL || r->connection->ssl == NULL) {
+        *err = "bad request";
+        return NGX_ERROR;
+    }
+
+    ssl_conn = r->connection->ssl->connection;
+    if (ssl_conn == NULL) {
+        *err = "bad ssl conn";
+        return NGX_ERROR;
+    }
+
+    c = ngx_ssl_get_connection(ssl_conn);
+
+    switch (c->local_sockaddr->sa_family) {
+        case AF_UNIX:
+            *err = "unix domain has no port";
+            return NGX_ERROR;
+        default:
+            p = ngx_inet_get_port(c->local_sockaddr);
+    }
+
+    if (p > 0 && p < 65536) {
+        *port = (unsigned int) p;
```

Comment:
I think if we found the `port` is `(p < 1 || p > 65535)`, we should return an error too?

Reply:
update

Reply:
I think like this the code will be more clean? 
```
if (port <1 || port > 65535) {
   // return error
}

//return port
```

Reply:
update